File: Structural/Decorator/JavaScript/coffeeDecorator.js
// coffeeDecorator.js

const Coffee = require("./coffee");

class CoffeeDecorator extends Coffee {
  constructor(coffee) {
    super();
    this.decoratedCoffee = coffee;
  }

  getDescription() {
    return this.decoratedCoffee.getDescription();
  }

  getCost() {
    return this.decoratedCoffee.getCost();
  }
}

module.exports = CoffeeDecorator;

File: Structural/Decorator/Python/coffee_decorator.py
# coffee_decorator.py

from coffee import Coffee

class CoffeeDecorator(Coffee):
  def __init__(self, coffee: Coffee):
    self._decorated_coffee = coffee

  def get_description(self) -> str:
    return self._decorated_coffee.get_description()

  def get_cost(self) -> float:
    return self._decorated_coffee.get_cost()
